src/phone_formatter.js

```diff
@@ -175,15 +217,37 @@ class PhoneFormatter extends DelimitedTextFormatter {
     }
 
     if (/^\d*$/.test(change.inserted.text) || change.proposed.text.indexOf('+') === 0) {
+      let formatName = this.delimiterMap.name;
+
+      // First guess at the localized format
+      if (currentCountryCode.localizedFormat) {
+        this.delimiterMap = currentCountryCode.localizedFormat;
+        this.maximumLength = currentCountryCode.localizedFormat.maximumLength;
+        formatName = `localized-${currentCountryCode.E164}`;
+      }
+
       // We need to store the change and current format guess so that if the isChangeValid
       // call to super changes the proposed text such that the format we thought is no longer
       // valid. If that does happen we actually just rerun it through with the correct format
+      const originalProposed = change.proposed.text;
       const _isChangeValid = super.isChangeValid(change, error);
-      const formatName = this.delimiterMap.name;
+
       this.guessFormatFromText(change.proposed.text);
-      return formatName === this.delimiterMap.name ?
-        _isChangeValid :
-        super.isChangeValid(change, error);
+
+      // Make sure if the localizedFormat changed, respect that
+      if (currentCountryCode.localizedFormat) {
+        this.delimiterMap = currentCountryCode.localizedFormat;
+        this.maximumLength = currentCountryCode.localizedFormat.maximumLength;
+        this.delimiterMap.name = `localized-${currentCountryCode.E164}`;
+      }
+
+      if (formatName === this.delimiterMap.name) {
```

Comment:
Sorry this is the downside to atomic fixes. This was merged in already to fix a bug. https://github.com/square/field-kit/blob/master/src/phone_formatter.js#L8